components/builder-api/src/server/resources/profile.rs

```diff
@@ -222,13 +215,54 @@ async fn set_license(req: HttpRequest,
                      state: Data<AppState>,
                      Json(payload): Json<LicensePayload>)
                      -> HttpResponse {
-    let conn = match state.db.get_conn().map_err(Error::DbError) {
+    let mut conn = match state.db.get_conn().map_err(Error::DbError) {
         Ok(conn_ref) => conn_ref,
         Err(err) => return err.into(),
     };
 
     match authorize_session(&req, None, None) {
         Ok(_session) => {
+            let base_url = &state.config.api.license_server_url;
```

Comment:
I'm not sure I understand what this added block that hits the license server is for. It downloads the license but doesn't seem to do anything and then calls `fetch_license_expiration` which would download the license again.